### ISSUE TITLE
Version 1.0.0+: Dependencies injected in lazy-loaded modules break page

### ISSUE THREAD

**Issue by Nudelsieb** [ocombe/ocLazyLoad#198]
This functionality worked before with v0.5.0. Unfortunately with the new versions it breaks.

In our project we are using packs that are loaded through a configuration json. This json defines the main plugins that are required for the pack. It works like a single application, which can be completely different with another configuration, hence other loaded plugins.

Those plugins have to be loaded by ocLazyLoad, which happens in ui-router like this:
```javascript
$ocLazyLoad.load(pluginsToLoad, {
	serie: true
})...
```
In some of the plug-ins we added dependencies like this:
```javascript
angular.module('angularProject.module1', [{
	name: 'plugin1',
	files: [
		'plugin1.module.js',
		'plugin1.model.js',
		'plugin1.controller.js',
		'plugin1.directive.js',
	],
	serie: true
}]);
```
Those modules may have dependencies set themselves. This creates a dependency chain.
However, now with ocLazyLoad 1.0.0+ this mechanism doesn't work anymore, despite mentioned in the readme.
There are a lot of errors which change randomly every page load, e.g.:
```javascript
[$injector:nomod] Module '[...]' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
```
or
```javascript
Error: no file to load has been found, if you're trying to load an existing module you should use the 'inject' method instead of 'load'.
```
To resolve this problem we have remove those dependency references from the modules and prepare the dependencies manually for each pack (in the pack's configuration json) in the right order for ozLazLoad to register them. This is obviously cumbersome and wastes a lot of time. I have no idea why the mechanism suddenly breaks.

Are we using ocLL wrong with chained dependencies?

**Comment by ocombe** [comment]
Hmm I think that it's probably my fault, I made a change in the code to avoid the case where some errors would not be rejected but obviously it was working in some cases.
Is there any chance that you could make me a minimal use case (on plnkr or something similar) to reproduce this problem please?

**Comment by atrusch** [comment]
Hi,

I am facing the same kind of problem. I have a plugin link to a state. When the state is active the modules are loaded but angular instanciate the template before all the module definition is parsed.
When I manually reload the page (F5) sometimes it works, sometimes not.

state definition 
{
	url : '/skills_user_admin',
	views : {
		'RootView' : {
			templateUrl : 'plugins/usl_userSkills/userSkillAdmin/usl_userSkillAdmin.html',
		}
	},
	resolve : {
		modules : ['$ocLazyLoad', function($ocLazyLoad) {
			return $ocLazyLoad.load([
        	   {type: 'css', path:'webjars/angular-ui-select/0.11.2/select.min.css'},
          	   {type: 'css', path:'js/thirdparty/angular/tree-grid-directive/treeGrid.css'},
        	   'webjars/angular-ui-select/0.11.2/select.min.js',
        	   'js/thirdparty/angular/tree-grid-directive/tree-grid-directive.js',
			   'plugins/usl_userSkills/usl_treeSkills.js',
			   'plugins/usl_userSkills/userSkillAdmin/usl_userSkillAdmin.js'
 			]);
		}]
		
	}
}


module definition 

"use strict";

angular.module('usl_userSkillsAdmin',["oc.lazyLoad", "ui.router", "restangular", "authorization", 'flashr', 
    'pascalprecht.translate', 'tmh.dynamicLocale', 'digLog', 'ngSanitize']);

angular.module('usl_userSkillsAdmin').run(function($ocLazyLoad){
	$ocLazyLoad.load({
		files : [
		    'plugins/usl_userSkills/userSkillAdmin/services/usl_userSkillAdminService.js'
		]
	})
});

angular.module('usl_userSkillsAdmin').controller('usl_userSkillsAdminCtrl',function($scope){
	$scope.lazyLoadDirective = {
			files:[ 
			    'plugins/usl_userSkills/userSkillAdmin/controllers/usl_userListCtrl.js'
			]
		};
});

template
<section class="content-header" ng-controller="usl_userSkillsAdminCtrl">
  <h1>
		{{'User skills' | translate}}
  </h1>
</section>

<section class="content" oc-lazy-load="{files:['plugins/usl_userSkills/userSkillAdmin/directives/usl_selectSkillLevel.js','plugins/usl_userSkills/userSkillAdmin/controllers/usl_userListCtrl.js']}">
	<div class="box box-default" ng-controller="usl_userListCtrl">
		<div class="box-header with-border">
			<h3 class="box-title ng-binding">{{'Details' | translate}}</h3>
			<div class="box-tools pull-right">
				<div class="col-xs-7">
					<ui-select ng-model="selected.user" theme="select2" style="width:300px" on-select="userSelected($item)">
						<ui-select-match placeholder="Select a person...">{{$select.selected.USR_FNAME}} {{$select.selected.USR_LNAME}}</ui-select-match>
						<ui-select-choices refresh="reload($select.search)" refresh-delay="200" repeat="person in people | propsFilter: {USR_LNAME: $select.search, USR_FNAME: $select.search}">
							<div ng-bind-html="person.USR_FNAME+' '+person.USR_LNAME | safehighlight: $select.search"></div>
							<small>
							email: {{person.USR_EMAIL}}
							</small>
						</ui-select-choices>
					</ui-select>
				</div>
			</div>
		</div>
		<div class="box-body">
			<tree-grid ng-show="canShowGrid" tree-data="treeData" icon-leaf="glyphicon glyphicon-education" tree-control="skill_tree" col-defs="gridConfig" expand-on="expanding_property"></tree-grid>
		</div>
	</div>
</section>



Error 
Error: [ng:areq] http://errors.angularjs.org/1.4.0/ng/areq?p0=usl_userSkillsAdminCtrl&p1=not%20aNaNunction%2C%20got%20undefined
    at Error (native)
    at http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:6:416
    at Sb (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:21:439)
    at Pa (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:22:25)
    at $get (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:79:173)
    at A (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:59:278)
    at M (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:60:121)
    at g (http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:54:300)
    at http://localhost.digora.com:8080/Portal/webjars/angularjs/1.4.0/angular.min.js:53:330
    at compile (http://localhost.digora.com:8080/Portal/webjars/angular-ui-router/0.2.14/angular-ui-router.min.js:7:22912)

**Comment by Nudelsieb** [comment]
Hi, thanks for the reply.

I've tried to setup a plnkr:
http://plnkr.co/edit/6iQJRQLscDzDiArXJzy9

Unfortunately, there is an error initialising ocLL: 
```javascript
Refused to execute script from 'https://raw.githubusercontent.com/ocombe/ocLazyLoad/1.0.1/dist/ocLazyLoad.js' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled.
angular.js:4361 Uncaught Error: [$injector:modulerr] Failed to instantiate module InjectorTest due to:
Error: [$injector:modulerr] Failed to instantiate module oc.lazyLoad due to:
Error: [$injector:nomod] Module 'oc.lazyLoad' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
```

Do you have any idea why this doesn't work at all in plnkr? The MIME type bit is very weird.

Thanks for your help.

**Comment by ocombe** [comment]
I think that I just fixed this issue with release 1.0.3, could you test if it works for you now ?

**Comment by Nudelsieb** [comment]
Hi ocombe, thanks for your hard work. I cannot easily reproduce the old setup we used. Anyway, we had to move one and work out something that will work in future setups. The new configuration involves requirejs with a bit oclazyload ($ocLazyLoad.inject) and it works great. Keep up the good work.
